From 9b34f562b2ea7ceb05830bf2e86351425b04268e Mon Sep 17 00:00:00 2001
From: EdGruberman <ed@rjump.com>
Date: Sat, 2 Feb 2013 14:34:23 -0700
Subject: [PATCH] Combination of portal fixes#
 This is a combination of fixes
 of pull reqs for Portal changes BUKKIT-3541, BUKKIT-3555, BUKKIT-3559,
 BUKKIT-3542

Plus my own fix to Eds patch about default PTA
---
 src/main/java/net/minecraft/server/PortalTravelAgent.java  | 12 ++++++++----
 src/main/java/org/bukkit/craftbukkit/CraftTravelAgent.java |  1 +
 2 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/src/main/java/net/minecraft/server/PortalTravelAgent.java b/src/main/java/net/minecraft/server/PortalTravelAgent.java
index cb5a0ad..0954b67 100644
--- a/src/main/java/net/minecraft/server/PortalTravelAgent.java
+++ b/src/main/java/net/minecraft/server/PortalTravelAgent.java
@@ -223,8 +223,10 @@ public class PortalTravelAgent {
                 int j3 = Direction.b[j2];
                 int k3 = Direction.a[l2];
                 int l3 = Direction.b[l2];
-                boolean flag1 = !this.a.isEmpty(i + i3 + k3, j, k + j3 + l3) || !this.a.isEmpty(i + i3 + k3, j + 1, k + j3 + l3);
-                boolean flag2 = !this.a.isEmpty(i + i3, j, k + j3) || !this.a.isEmpty(i + i3, j + 1, k + j3);
+                // CraftBukkit start - check for blocks 2 up as well
+                boolean flag1 = !this.a.isEmpty(i + i3 + k3, j, k + j3 + l3) || !this.a.isEmpty(i + i3 + k3, j + 1, k + j3 + l3) || !this.a.isEmpty(i + i3 + k3, j + 2, k + j3 + l3);
+                boolean flag2 = !this.a.isEmpty(i + i3, j, k + j3) || !this.a.isEmpty(i + i3, j + 1, k + j3) || !this.a.isEmpty(i + i3, j + 2, k + j3);
+                // CraftBukkit end
 
                 if (flag1 && flag2) {
                     j2 = Direction.f[j2];
@@ -238,8 +240,10 @@ public class PortalTravelAgent {
                     int i4 = k - l3;
 
                     d4 -= (double) l3;
-                    flag1 = !this.a.isEmpty(k1 + i3 + k3, j, i4 + j3 + l3) || !this.a.isEmpty(k1 + i3 + k3, j + 1, i4 + j3 + l3);
-                    flag2 = !this.a.isEmpty(k1 + i3, j, i4 + j3) || !this.a.isEmpty(k1 + i3, j + 1, i4 + j3);
+                    // CraftBukkit start - check for blocks 2 up as well
+                    flag1 = !this.a.isEmpty(k1 + i3 + k3, j, i4 + j3 + l3) || !this.a.isEmpty(k1 + i3 + k3, j + 1, i4 + j3 + l3) || !this.a.isEmpty(k1 + i3 + k3, j + 2, i4 + j3 + l3);
+                    flag2 = !this.a.isEmpty(k1 + i3, j, i4 + j3) || !this.a.isEmpty(k1 + i3, j + 1, i4 + j3) || !this.a.isEmpty(k1 + i3, j + 2, i4 + j3);
+                    // CraftBukkit end
                 }
 
                 float f1 = 0.5F;
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftTravelAgent.java b/src/main/java/org/bukkit/craftbukkit/CraftTravelAgent.java
index fad2d8a..0ab71cb 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftTravelAgent.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftTravelAgent.java
@@ -1,6 +1,7 @@
 package org.bukkit.craftbukkit;
 
 import net.minecraft.server.ChunkCoordinates;
+import net.minecraft.server.MinecraftServer;
 import net.minecraft.server.PortalTravelAgent;
 import net.minecraft.server.WorldServer;
 
-- 
1.8.1.1

